Common/src/Common/Service/Data/Search/SearchTypeManager.php

```diff
@@ -23,4 +23,28 @@ class SearchTypeManager extends AbstractPluginManager
      * @var bool
      */
     protected $autoAddInvokableClass = false;
+
+    private array $config;
+
+    public function __construct(ContainerInterface $container, $config)
+    {
+        $this->config = $config;
+        parent::__construct($container, $config);
+    }
+
+    /**
+     * @todo this isn't great, but provides for backward compatibility with old code as we move to Laminas 3
+     *
+     * the old code (mis)used a debug method within the Laminas service manager to provide this info,
+     * for now we're replicating that behaviour here
+     */
```

Comment:
Not sure I understand this. Do we need to provide backwards compatibility here (or anyway anymore)?

Reply:
Just a poor choice of words...basically it just gives the search code what it got previously from the removed Laminas 2 ServiceManager method...without having to risk altering the search code itself. The @todo is there because I'd assumed at the first opportunity we'd re-engineer it.